From f2a700834f877c8422fe87dc9aead97a190ca1d2 Mon Sep 17 00:00:00 2001
From: Arnon Kaufman <akaufman@habana.ai>
Date: Mon, 08 Jul 2024 18:56:34 +0300
Subject: [PATCH] [SW-15774] board: adjust intermediate bins dirs

Use the target-arch bins directory for intermediate binary files.

Change-Id: I572414b426c5465b491dae0135492937604c6061
Signed-off-by: Arnon Kaufman <akaufman@habana.ai>
---

diff --git a/board/habanalabs/fs1_arc/post-image-fit.sh b/board/habanalabs/fs1_arc/post-image-fit.sh
index 34524d4..449d86a 100755
--- a/board/habanalabs/fs1_arc/post-image-fit.sh
+++ b/board/habanalabs/fs1_arc/post-image-fit.sh
@@ -20,10 +20,10 @@
 
 	# copy binaries from other HL targets required for boot fit creation
 	# TEMPORARY: arcpid isn't ready yet, so comment-out the following lines. Otherwise, the script fails.
-	#cp ${BR2_EXTERNAL}/output/${DEV_NAME}_arc/images/${DEV_PREFIX}-zephyr-pid-iccm.bin ${FILEPATH}/${DEV_PREFIX}-zephyr-pid-iccm.bin.le
-	#cp ${BR2_EXTERNAL}/output/${DEV_NAME}_arc/images/${DEV_PREFIX}-zephyr-pid-dccm.bin ${FILEPATH}/${DEV_PREFIX}-zephyr-pid-dccm.bin.le
-	cp ${BR2_EXTERNAL}/output/${DEV_NAME}_arc/images/${DEV_PREFIX}-zephyr-mgmt-iccm.bin ${FILEPATH}/${DEV_PREFIX}-zephyr-mgmt-iccm.bin.le
-	cp ${BR2_EXTERNAL}/output/${DEV_NAME}_arc/images/${DEV_PREFIX}-zephyr-mgmt-dccm.bin ${FILEPATH}/${DEV_PREFIX}-zephyr-mgmt-dccm.bin.le
+	cp ${BINS_DIR}/${DEV_PREFIX}-zephyr-pid-iccm.bin ${FILEPATH}/${DEV_PREFIX}-zephyr-pid-iccm.bin.le
+	cp ${BINS_DIR}/${DEV_PREFIX}-zephyr-pid-dccm.bin ${FILEPATH}/${DEV_PREFIX}-zephyr-pid-dccm.bin.le
+	cp ${BINS_DIR}/${DEV_PREFIX}-zephyr-mgmt-iccm.bin ${FILEPATH}/${DEV_PREFIX}-zephyr-mgmt-iccm.bin.le
+	cp ${BINS_DIR}/${DEV_PREFIX}-zephyr-mgmt-dccm.bin ${FILEPATH}/${DEV_PREFIX}-zephyr-mgmt-dccm.bin.le
 
 	return
 }
diff --git a/board/habanalabs/gaudi3/post-image-fit.sh b/board/habanalabs/gaudi3/post-image-fit.sh
index ecb110d..df61152 100644
--- a/board/habanalabs/gaudi3/post-image-fit.sh
+++ b/board/habanalabs/gaudi3/post-image-fit.sh
@@ -52,17 +52,17 @@
 	FILEPATH=$1
 
 	# copy binaries from other HL targets required for boot fit creation
-	cp ${STAGE_DIR}/${DEV_PREFIX}-zephyr-pid-iccm.bin ${FILEPATH}/${DEV_PREFIX}-zephyr-pid-iccm.bin.le
-	cp ${STAGE_DIR}/${DEV_PREFIX}-zephyr-pid-dccm.bin ${FILEPATH}/${DEV_PREFIX}-zephyr-pid-dccm.bin.le
-	cp ${STAGE_DIR}/${DEV_PREFIX}-zephyr-mgmt-iccm.bin ${FILEPATH}/${DEV_PREFIX}-zephyr-mgmt-iccm.bin.le
-	cp ${STAGE_DIR}/${DEV_PREFIX}-zephyr-mgmt-dccm.bin ${FILEPATH}/${DEV_PREFIX}-zephyr-mgmt-dccm.bin.le
+	cp ${BINS_DIR}/${DEV_PREFIX}-zephyr-pid-iccm.bin ${FILEPATH}/${DEV_PREFIX}-zephyr-pid-iccm.bin.le
+	cp ${BINS_DIR}/${DEV_PREFIX}-zephyr-pid-dccm.bin ${FILEPATH}/${DEV_PREFIX}-zephyr-pid-dccm.bin.le
+	cp ${BINS_DIR}/${DEV_PREFIX}-zephyr-mgmt-iccm.bin ${FILEPATH}/${DEV_PREFIX}-zephyr-mgmt-iccm.bin.le
+	cp ${BINS_DIR}/${DEV_PREFIX}-zephyr-mgmt-dccm.bin ${FILEPATH}/${DEV_PREFIX}-zephyr-mgmt-dccm.bin.le
 
 	# TODO: cleanup once transition to dccm offload is complete
-	[[ -f ${STAGE_DIR}/${DEV_PREFIX}-zephyr-mgmt-offload.bin ]] \
-		&& cp ${STAGE_DIR}/${DEV_PREFIX}-zephyr-mgmt-offload.bin ${FILEPATH}/${DEV_PREFIX}-zephyr-mgmt-offload.bin.le
+	[[ -f ${BINS_DIR}/${DEV_PREFIX}-zephyr-mgmt-offload.bin ]] \
+		&& cp ${BINS_DIR}/${DEV_PREFIX}-zephyr-mgmt-offload.bin ${FILEPATH}/${DEV_PREFIX}-zephyr-mgmt-offload.bin.le
 
-	cp ${STAGE_DIR}/${DEV_PREFIX}-zephyr-event-iccm.bin ${FILEPATH}/${DEV_PREFIX}-zephyr-event-iccm.bin.le
-	cp ${STAGE_DIR}/${DEV_PREFIX}-zephyr-event-dccm.bin ${FILEPATH}/${DEV_PREFIX}-zephyr-event-dccm.bin.le
+	cp ${BINS_DIR}/${DEV_PREFIX}-zephyr-event-iccm.bin ${FILEPATH}/${DEV_PREFIX}-zephyr-event-iccm.bin.le
+	cp ${BINS_DIR}/${DEV_PREFIX}-zephyr-event-dccm.bin ${FILEPATH}/${DEV_PREFIX}-zephyr-event-dccm.bin.le
 
 	prep_eth_phy_fw_binary ${ETH_PHY_FW_HEX_FILE} ${ETH_PHY_FW_BIN_FILE}
 
